File: py/IEC61968/Work/Vehicle.py
from IEC61968.Work.VehicleUsageKind import VehicleUsageKind
from IEC61968.Work.WorkAsset import WorkAsset
from IEC61970.Base.Domain.DateTime import DateTime
from IEC61970.Base.Domain.Length import Length


class Vehicle(WorkAsset):
    """
    Vehicle asset.
    @created 29-Dec-2023 5:41:47 PM
    """

    def __init__(self):
        super().__init__()

        # Date and time the last odometer reading was recorded.
        self.odometer_read_date_time = DateTime()

        # Odometer reading of this vehicle as of the 'odometer_reading_date_time'. Refer to
        # associated ActivityRecords for earlier readings.
        self.odometer_reading = Length()

        # Kind of usage of the vehicle.
        self.usage_kind = VehicleUsageKind.CREW

File: java/IEC61968/Work/Vehicle.java
package IEC61968.Work;

import IEC61970.Base.Domain.DateTime;
import IEC61970.Base.Domain.Length;

/**
 * Vehicle asset.
 * @created 02-Jan-2024 10:07:00 PM
 */
public class Vehicle extends WorkAsset {

	/**
	 * Date and time the last odometer reading was recorded.
	 */
	public DateTime odometerReadDateTime;
	/**
	 * Odometer reading of this vehicle as of the 'odometerReadingDateTime'. Refer to
	 * associated ActivityRecords for earlier readings.
	 */
	public Length odometerReading;
	/**
	 * Kind of usage of the vehicle.
	 */
	public VehicleUsageKind usageKind;

	public Vehicle(){

	}
}//end Vehicle